fetch.js

```diff
@@ -195,44 +195,40 @@
       }
     }
 
-    if (support.blob) {
-      this.blob = function() {
-        var rejected = consumed(this)
-        if (rejected) {
-          return rejected
-        }
 
-        if (this._bodyBlob) {
-          return Promise.resolve(this._bodyBlob)
-        } else if (this._bodyFormData) {
-          throw new Error('could not read FormData body as blob')
-        } else {
-          return Promise.resolve(new Blob([this._bodyText]))
-        }
+    this.blob = function() {
+      var rejected = consumed(this)
+      if (rejected) {
+        return rejected
       }
 
-      this.arrayBuffer = function() {
-        return this.blob().then(readBlobAsArrayBuffer)
+      if (this._bodyBlob) {
+        return Promise.resolve(this._bodyBlob)
+      } else if (this._bodyFormData) {
+        throw new Error('could not read FormData body as blob')
+      } else {
+        return new Promise(function(resolve){
+          resolve(new Blob([this._bodyText]))
```

Comment:
Potato tomato when i wrote this i thought rejection ^^
I thought yea, it should always return a Promise... You must handle consumed either way also

When you are writing it like this it will end up doing the same thing
```javascript
fetch('/').then(response => response.blob()).catch(error => …) // Blob is undefined error
```

but what if you wrote it like this
```javascript
res = new Response('x')
res.text()
try {
res.blob().catch(err => {
  // body have been used
})
} catch(err) {
  // blob not supported
}
```